setup.js

```diff
@@ -0,0 +1,106 @@
+var fs = require('fs');
+var prompt = require('prompt');
+var template = require('lodash.template');
+var merge = require('lodash.merge');
+var rmraf = require('rimraf');
+
+var argv = process.argv;
+
+var defaults = {
+  version: '0.0.1',
+  yourName: 'We Are Fractal',
+  yourEmail: 'contact@wearefractal.com',
+  yourDomain: 'http://wearefractal.com',
+  gitUserName: 'wearefractal',
+  main: './index.js'
+};
+
+
+var questions = {
+  properties: {
+    projectName: {
+      pattern: /^[a-zA-Z\-]+$/,
+      message: 'name must not contain spaces',
+      required: true
+    },
+    description: {},
+    version: {
+      default: defaults.version
+    },
+    yourName: {
+      required: true,
+      default: defaults.yourName
+    },
+    yourEmail: {
+      required: true,
+      default: defaults.yourEmail
+    },
+    yourDomain: {
+      required: true,
+      default: defaults.yourDomain
+    },
+    gitUserName: {
+      required: true,
+      default: defaults.gitUserName
+    },
+    main: {
+      default: defaults.main
+    }
+  }
+};
+
+prompt.start();
+
+if (argv[2] == '--use-defaults') {
+  console.log('Using defaults \n');
+
+  var defaultQuestions = {
+    properties: {
+      projectName: {
+        pattern: /^[a-zA-Z\-]+$/,
+        message: 'name must not contain spaces',
+        required: true
+      },
+      description: {},
+    }
+  };
+
+  return prompt.get(defaultQuestions, function(err, data) {
+    data = merge(data, defaults);
+    save(data);
+  });
+}
+
+prompt.get(questions, function(err, data) {
+  save(data);
+});
+
+
+function save(data) {
+  if (data === null || data === undefined) {
+    return;
+  }
+  data.year = new Date().getFullYear();
+  var pkg = './package.json';
```

Comment:
you should split all of this logic out into a function that renames and templates, then make an array of files to template and forEach it through that function